### ISSUE TITLE
No clear route to add/update a new row.

### ISSUE THREAD

**Issue by JosepAlacid** [assuncaocharles/ngx-indexed-db#275]
When trying to update a row in the DB using:
```
this.localDB.update("storeName", newRow).subscribe(
   //next
   (result) => console.log(result),
   //error
   (error) => {
      this.localDB.add("storeName", newRow).subscribe(
         //next
         (result) => console.log(result),
         () => console.log("unable to update")
      )
   }
);
```
If the DB is empty or the key exists, there's no issue. 
But if there are any other rows, then the runtime error isn't captured and console shows only:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'isStopped')
```

**Comment by aparzi** [comment]
Hi @JosepAlacid,
the problem is still present? If so the request will be taken care of.

P.s. sorry if it's been some time to get back to you

**Comment by JosepAlacid** [comment]
Hi Angelo!

I've been tracing your code a bit and, as it seems to me, some other library (RxJs, Zone) is catching your error messages before they get to the console. 
I will try to follow that thread again and post more detailed info.

P.S. Your PS is very kind!

**Comment by aparzi** [comment]
Hi @JosepAlacid,
I tried to test your code and it seems like there is no problem. Can you give more information about it, for example by sharing the whole function or the code?
Also, I don't understand why you don't just use the update method to add/update a row. I would like to remind you that if the record is not present it is still inserted into the DB.

**Comment by JosepAlacid** [comment]
It seems any error further information is blocked here:
```
function optionsGenerator(type, storeName, reject, resolve) {
    return {
        storeName,
        dbMode: type,
        error: (e) => {
            reject(e);
        },
        abort: (e) => {
            reject(e);
        },
    };
}
```
I was tracing the code but after `reject(e)` error is logged in the console there is no other explanation than `Uncaught TypeError: Cannot read properties of undefined (reading 'isStopped')` shown. It's a hard way to debug if no explanation than this is offered.

Some other people are experiencing this issue (https://es.stackoverflow.com/questions/482254/ngx-indexed-db-no-captura-error-al-subscribirse) but some of them just took any workaround.

I'm sorry that I can't be more helpful so far. I've never used pull requests and it will take me some time to get into it.

**Comment by assuncaocharles** [comment]
Hi @JosepAlacid I believe this got fixed together in #319 

Therefore I am closing this, feel free to reopen it if you find a different problem.
